### PR TITLE
nafis register display

### PR DESCRIPTION
**Pull request by al-nafis**
Created Register Display

### CODE REVIEW

File: app/src/main/res/layout/activity_register.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        android:id="@+id/et_register_first_name"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintBottom_toTopOf="@+id/et_register_last_name"
+        android:hint="@string/prompt_first_name"
+        android:inputType="textPersonName"
+        android:maxLines="1"
```
Comment (itzcoatl90) [inline app/src/main/res/layout/activity_register.xml]: nop.... no magic numbers... send this to values resource .xml

File: app/src/main/res/layout/activity_register.xml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        android:id="@+id/et_register_last_name"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toBottomOf="@id/et_register_first_name"
+        app:layout_constraintBottom_toTopOf="@+id/et_register_email"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:hint="@string/prompt_last_name"
+        android:inputType="textPersonName"
+        android:maxLines="1"
```
Comment (itzcoatl90) [inline app/src/main/res/layout/activity_register.xml]: Same here

File: app/src/main/res/layout/activity_register.xml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        android:id="@+id/et_register_email"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toBottomOf="@id/et_register_last_name"
+        app:layout_constraintBottom_toTopOf="@+id/et_register_city"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:hint="@string/prompt_email"
+        android:inputType="textEmailAddress"
+        android:maxLines="1"
```
Comment (itzcoatl90) [inline app/src/main/res/layout/activity_register.xml]: Same

File: app/src/main/res/layout/activity_register.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        android:id="@+id/et_register_city"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toBottomOf="@id/et_register_email"
+        app:layout_constraintBottom_toTopOf="@+id/et_register_password"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:hint="@string/prompt_city"
+        android:inputType="textNoSuggestions"
+        android:maxLines="1"
```
Comment (itzcoatl90) [inline app/src/main/res/layout/activity_register.xml]: here

File: app/src/main/res/layout/activity_register.xml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        android:id="@+id/et_register_password"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toBottomOf="@id/et_register_city"
+        app:layout_constraintBottom_toTopOf="@+id/et_register_retype_password"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:hint="@string/prompt_password"
+        android:inputType="textPassword"
+        android:maxLines="1"
```
Comment (itzcoatl90) [inline app/src/main/res/layout/activity_register.xml]: again

File: app/src/main/res/layout/activity_register.xml [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        android:id="@+id/et_register_retype_password"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toBottomOf="@id/et_register_password"
+        app:layout_constraintBottom_toTopOf="@+id/btn_register_registerBtn"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:hint="@string/prompt_retype_password"
+        android:inputType="textPassword"
+        android:maxLines="1"
```
Comment (itzcoatl90) [inline app/src/main/res/layout/activity_register.xml]: nop

**Review by itzcoatl90 — CHANGES_REQUESTED** [review]
Attend comments and also please solve merge conflicts...

File: app/.gitignore [hunk 7]
```diff
@@ -1 +1,10 @@
+*.iml
```
Comment (itzcoatl90) [inline app/.gitignore:1]: Good job

**Review by itzcoatl90 — CHANGES_REQUESTED** [review]
lol... almost got it! No comments on the files changes, but still having merge conflicts.... solve them before putting ticket into "PR Review"

File: app/.gitignore (hunk 7 above)
Comment (al-nafis) [reply app/.gitignore:1]: Is that something I shouldn't have on gitignore??

**Comment by al-nafis** [comment]
I merged the conflicts. But confused about +*.iml in gitignore file. Is this something I should delete from the ignore file?

File: app/.gitignore (hunk 7 above)
Comment (itzcoatl90) [reply app/.gitignore:1]: I meant Good job.... lol... I think "good job" is a good thing right??  Good job in the entire file

File: app/build.gradle [hunk 8]
```diff
@@ -18,11 +18,16 @@ android {
     }
 }
 
+repositories {
+    google()
+}
+
 dependencies {
+    compile 'com.android.support:design:27.1.1'
```
Comment (itzcoatl90) [inline app/build.gradle]: noooooo!!!!!     I though we were gonna merge it today... :S    Please change compile to implementation... in my understanding, "compile" is deprecated now

File: app/build.gradle [hunk 9]
```diff
@@ -18,11 +18,16 @@ android {
...
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'com.android.support:appcompat-v7:27.1.1'
     implementation 'com.android.support.constraint:constraint-layout:1.1.2'
-    implementation 'com.android.support:design:27.1.1'
```
Comment (itzcoatl90) [inline app/build.gradle]: That means not erasing this line of code

File: app/build.gradle [hunk 10]
```diff
@@ -18,11 +18,16 @@ android {
...
+
```
Comment (itzcoatl90) [inline app/build.gradle]: And try to avoid empty lines in PRs as we want PR as clean as possible.... tomorrow I will talk about cleanliness on your PRs

**Review by itzcoatl90 — CHANGES_REQUESTED** [review]
Almost have it

File: app/.gitignore (hunk 7 above)
Comment (al-nafis) [reply app/.gitignore:1]: lol

**Comment by al-nafis** [comment]
Changed compile to implementation 'com.android.support:design:27.1.1'

File: app/build.gradle [hunk 11]
```diff
@@ -18,11 +18,14 @@ android {
...
 
+repositories {
+    google()
+}
+
 dependencies {
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation 'com.android.support:appcompat-v7:27.1.1'
     implementation 'com.android.support.constraint:constraint-layout:1.1.2'
-    implementation 'com.android.support:design:27.1.1'
```
Comment (itzcoatl90) [inline app/build.gradle]: Are you sure erasing this line of code won't break the code somewhere?

**Review by itzcoatl90 — CHANGES_REQUESTED** [review]
:( The review passed.... but your branch is breaking something... I cannot build the project in my local with your branch... are you sure in yours is working correctly?

Hint... the issue is in the string.xml

**Comment by itzcoatl90** [comment]
screen shot 2018-07-20 at 12 28 23 am

**Comment by al-nafis** [comment]
Can you see now? 
I merged the conflicts in android studio and then pushed it.

File: .idea/codeStyles/Project.xml [hunk 12]
```diff
@@ -0,0 +1,29 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (itzcoatl90) [inline .idea/codeStyles/Project.xml:1]: What is this file? it should not be pushed... clean your configuration

File: .idea/vcs.xml [hunk 13]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (itzcoatl90) [inline .idea/vcs.xml:1]: This file should not appear in PR nor the checksums.ser

File: app/build.gradle (hunk 11 above)
Comment (itzcoatl90) [reply app/build.gradle]: Do not erase this line

File: app/src/main/res/values/dimens.xml [hunk 14]
```diff
@@ -9,10 +9,7 @@
     <dimen name="te_button_height">45dp</dimen>
     <dimen name="views_update_top_margin">100dp</dimen>
     <dimen name="views_update_bottom_margin">150dp</dimen>
-
-    <dimen name="input_margin_start_to_end">8dp</dimen>
-    <dimen name="activity_horizontal_margin">8dp</dimen>
-    <dimen name="activity_vertical_margin">8dp</dimen>
```
Comment (itzcoatl90) [inline app/src/main/res/values/dimens.xml]: Not erasing lines of code

File: app/src/main/res/values/strings.xml [hunk 15]
```diff
@@ -12,5 +23,5 @@
     <string name="retype_password">Retype Password</string>
     <string name="tv_update_title">Update Password</string>
     <string name="button_submit">Submit</string>
-
```
Comment (itzcoatl90) [inline app/src/main/res/values/strings.xml:15]: no erasing lines of code

File: app/src/main/res/values/styles.xml [hunk 16]
```diff
@@ -1,19 +1,16 @@
...
-
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
+    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
         <!-- Customize your theme here. -->
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
         <item name="colorAccent">@color/colorAccent</item>
     </style>
-
```
Comment (itzcoatl90) [inline app/src/main/res/values/styles.xml:10]: no

File: app/src/main/res/values/styles.xml [hunk 17]
```diff
@@ -1,19 +1,16 @@
...
     <style name="AppTheme.NoActionBar">
         <item name="windowActionBar">false</item>
         <item name="windowNoTitle">true</item>
     </style>
-
```
Comment (itzcoatl90) [inline app/src/main/res/values/styles.xml:15]: no

File: app/src/main/res/values/styles.xml [hunk 18]
```diff
@@ -1,19 +1,16 @@
...
     <style name="AppTheme.AppBarOverlay" parent="ThemeOverlay.AppCompat.Dark.ActionBar" />
-
```
Comment (itzcoatl90) [inline app/src/main/res/values/styles.xml]: no

File: app/src/main/res/values/styles.xml [hunk 19]
```diff
@@ -1,19 +1,16 @@
...
     <style name="AppTheme.PopupOverlay" parent="ThemeOverlay.AppCompat.Light" />
-</resources>
```
Comment (itzcoatl90) [inline app/src/main/res/values/styles.xml]: This is not a change, it shouldn't appear in PR

**Pull request closed without merging**